temporal-sdk/src/main/java/io/temporal/common/context/OpenTracingContextPropagator.java

```diff
@@ -0,0 +1,207 @@
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+/** Support for OpenTracing spans */
+public class OpenTracingContextPropagator implements ContextPropagator {
+
+  private static final Logger log = LoggerFactory.getLogger(OpenTracingContextPropagator.class);
+
+  private static final ThreadLocal<SpanContext> currentOpenTracingSpanContext = new ThreadLocal<>();
+  private static final ThreadLocal<Span> currentOpenTracingSpan = new ThreadLocal<>();
+  private static final ThreadLocal<Scope> currentOpenTracingScope = new ThreadLocal<>();
+
+  public static SpanContext getCurrentOpenTracingSpanContext() {
+    return currentOpenTracingSpanContext.get();
+  }
+
+  public static void setCurrentOpenTracingSpanContext(SpanContext ctx) {
+    if (ctx != null) {
+      currentOpenTracingSpanContext.set(ctx);
+    }
+  }
+
+  @Override
+  public String getName() {
+    return "OpenTracing";
+  }
+
+  @Override
+  public Map<String, Payload> serializeContext(Object context) {
+    Map<String, Payload> serializedContext = new HashMap<>();
+    Map<String, String> contextMap = (Map<String, String>) context;
+    if (contextMap != null) {
+      for (Map.Entry<String, String> entry : contextMap.entrySet()) {
+        serializedContext.put(
+            entry.getKey(), DataConverter.getDefaultInstance().toPayload(entry.getValue()).get());
+      }
+    }
+    return serializedContext;
+  }
+
+  @Override
+  public Object deserializeContext(Map<String, Payload> context) {
+    Map<String, String> contextMap = new HashMap<>();
+    for (Map.Entry<String, Payload> entry : context.entrySet()) {
+      contextMap.put(
+          entry.getKey(),
+          DataConverter.getDefaultInstance()
+              .fromPayload(entry.getValue(), String.class, String.class));
+    }
+    return contextMap;
+  }
+
+  @Override
+  public Object getCurrentContext() {
+    Tracer currentTracer = GlobalTracer.get();
+    Span currentSpan = currentTracer.scopeManager().activeSpan();
+    if (currentSpan != null) {
+      HashMapTextMap contextTextMap = new HashMapTextMap();
+      currentTracer.inject(currentSpan.context(), Format.Builtin.TEXT_MAP, contextTextMap);
+      return contextTextMap.getBackingMap();
+    } else {
+      return null;
+    }
+  }
+
+  @Override
+  public void setCurrentContext(Object context) {
+    Tracer currentTracer = GlobalTracer.get();
+    Map<String, String> contextAsMap = (Map<String, String>) context;
+    if (contextAsMap != null) {
+      HashMapTextMap contextTextMap = new HashMapTextMap(contextAsMap);
+      setCurrentOpenTracingSpanContext(
+          currentTracer.extract(Format.Builtin.TEXT_MAP, contextTextMap));
+    }
+  }
+
+  @Override
+  public void setUp() {
+    Tracer openTracingTracer = GlobalTracer.get();
+    Tracer.SpanBuilder builder = openTracingTracer.buildSpan("cadence.workflow");
```

Comment:
I don't think it is going to work. The problem is that during replay a new span is going to be created. 
Unfortunately, there is no way to create a span passing deterministically generated ID to it. 

Another option would be using [SideEffect](https://github.com/temporalio/sdk-java/blob/4bb23c3a660de6a2630901d0dc6c049d9742ebe1/temporal-sdk/src/main/java/io/temporal/workflow/Workflow.java#L920) to store the span, but that implies additional MarkerEvent for every workflow.

Reply:
Does it really matter if we start multiple spans if we only finish one?  I don't think spans get reported until you finish() them.